### ISSUE TITLE
rclone pads files to 4096 with 0x00 when copying to WSL2 disk

### ISSUE THREAD

**Issue by BarsMonster** [rclone/rclone#6801]
#### What is the problem you are having with rclone?
When rclone writes files to windows WSL2 folder - it pads all files to 4096 byte boundary with 0x00. 
It detects that size is incorrect and skip the file. If you force it to continue with --ignore-size - you will receive files padded with 0's (i.e. almost all files will be damaged during operation). 

This behavior is only observed in rclone, text editors / total commander write/edit files without padding. Behavior is the same if direct WSL2 path is used, without subst. Not posting on forum as it looks very rare and clearly broken. 

#### What is your rclone version (output from `rclone version`)
rclone v1.61.1
- os/version: Microsoft Windows 11 Pro 22H2 (64 bit)
- os/kernel: 10.0.22621.1265 (x86_64)
- os/type: windows
- os/arch: 386
- go/version: go1.19.4
- go/linking: static
- go/tags: cmount

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Microsoft Windows 11 Pro 22H2 (64 bit)

#### Which cloud storage system are you using? (e.g. Google Drive)
None, local copy

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
rclone copy f:\test\ m:\data\test\ -P --ignore-size -vv
#Here F is normal local windows disk.  M is local WSL2 folder mapped to a directory through "subst" command. 

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```c:\my\tools\rclone>rclone copy f:\test\ m:\data\test\ -P --ignore-size -vv
2023/03/03 04:18:10 DEBUG : rclone: Version "v1.61.1" starting with parameters ["rclone" "copy" "f:\\test\\" "m:\\data\\test\\" "-P" "--ignore-size" "-vv"]
2023/03/03 04:18:10 DEBUG : Creating backend with remote "f:\\test\\"
2023/03/03 04:18:10 DEBUG : Using config file from "C:\\Users\\M\\.config\\rclone\\rclone.conf"
2023/03/03 04:18:10 DEBUG : fs cache: renaming cache item "f:\\test\\" to be canonical "//?/f:/test"
2023/03/03 04:18:10 DEBUG : Creating backend with remote "m:\\data\\test\\"
2023/03/03 04:18:10 DEBUG : fs cache: renaming cache item "m:\\data\\test\\" to be canonical "//?/m:/data/test"
2023-03-03 04:18:10 DEBUG : Local file system at //?/m:/data/test: Waiting for checks to finish
2023-03-03 04:18:10 DEBUG : Local file system at //?/m:/data/test: Waiting for transfers to finish
2023-03-03 04:18:10 DEBUG : test.txt: md5 = a5ea0ad9260b1550a14cc58d2c39b03d OK
2023-03-03 04:18:10 INFO  : test.txt: Copied (new)
Transferred:              3 B / 3 B, 100%, 0 B/s, ETA -
Transferred:            1 / 1, 100%
Elapsed time:         0.9s
2023/03/03 04:18:10 INFO  :
Transferred:              3 B / 3 B, 100%, 0 B/s, ETA -
Transferred:            1 / 1, 100%
Elapsed time:         0.9s

2023/03/03 04:18:10 DEBUG : 3 go routines active

c:\my\tools\rclone>dir m:\data\test
 Volume in drive M has no label.

 Directory of m:\data\test

03-Mar-23  03:53             4'096 test.txt
03-Mar-23  04:18    <DIR>          ..
03-Mar-23  04:18    <DIR>          .
               1 File(s)          4'096 bytes
               2 Dir(s)  8'534'423'404'544 bytes free
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by BarsMonster** [comment]
![padding](https://user-images.githubusercontent.com/4014665/222624377-2d70ef93-3690-479a-b772-c6056acdb6ce.png)

**Comment by ncw** [comment]
> When rclone writes files to windows WSL2 folder - it pads all files to 4096 byte boundary with 0x00.
> It detects that size is incorrect and skip the file. If you force it to continue with --ignore-size - you will receive files padded with 0's (i.e. almost all files will be damaged during operation).

This is probably a Windows bug tickled by rclone's use of pre-allocation - that would be my guess.

Try adding this and see if it helps.

      --local-no-preallocate   Disable preallocation of disk space for transferred files

**Comment by BarsMonster** [comment]
I can confirm that --local-no-preallocate is a valid workaround here.

**Comment by ncw** [comment]
Preallocate and sparsing are probably both lesser used features of file systems in general.

I think that then don't work or produce corrupted files is a bug in Windows rather than rclone alas.

**Comment by Masamune3210** [comment]
Yeah, I'm guessing they have a bug in either how they handle fallocate or their translation from the Linux world to the windows world when it comes to preallocation, because I've seen this bug manifest in other repos as well

**Comment by ncw** [comment]
I'm going to close this as it isn't a bug in rclone and there is a workaround above.
